js/packages/teams-ai/src/StreamingResponse.ts

```diff
@@ -100,43 +105,117 @@ export class StreamingResponse {
             throw new Error('The stream has already ended.');
         }
 
-        // Send final message
+        // Queue final message
         this._ended = true;
-        return this.sendActivity('message', this._message, {
-            streamType: 'final',
-            streamSequence: this._nextSequence++
+        this.queueNextChunk();
+
+        // Wait for the queue to drain
+        return this._queueSync!;
+    }
+
+    /**
+     * Waits for the outgoing activity queue to be empty.
+     * @returns {Promise<void>} - A promise representing the async operation.
+     */
+    public waitForQueue(): Promise<void> {
+        return this._queueSync || Promise.resolve();
+    }
+
+    /**
+     * Queues the next chunk of text to be sent to the client.
+     * @private 
+     */
+    private queueNextChunk(): void {
```

Comment:
each delta that openAI currently returns is a token, could we explore grouping these tokens into a chunk before sending? Confused how we're not hitting a throttle right now but Bot API has a limit of 1 RPS - this might be a problem in the future